includes/Core/Util/HelperUpdater.php

```diff
@@ -0,0 +1,196 @@
+			'author'        => '<a href="https://opensource.google.com">Google</a>',
+			'download_link' => $data['download_url'],
+			'trunk'         => $data['download_url'],
+			'tested'        => $data['tested'],
+			'requires'      => $data['requires'],
+			'requires_php'  => $data['requires_php'],
+			'last_updated'  => $data['last_updated'],
+		);
+		if ( ! empty( $data['icons'] ) ) {
+			$new_data['icons'] = $data['icons'];
+		}
+		if ( ! empty( $data['banners'] ) ) {
+			$new_data['banners'] = $data['banners'];
+		}
+		if ( ! empty( $data['banners_rtl'] ) ) {
+			$new_data['banners_rtl'] = $data['banners_rtl'];
+		}
+
+		return (object) $new_data;
+	}
+
+	/**
+	 * Retrieves plugin update data from the Site Kit REST API.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param object $value Update check object.
+	 * @return object Modified update check object.
+	 */
+	private function updater_data( $value ) {
+
+		// Stop here if the current user does not have sufficient capabilities.
+		if ( ! current_user_can( 'update_plugins' ) ) {
+			return $value;
+		}
+
+		$data = $this->fetch_plugin_data();
+
+		if ( empty( $data['download_url'] ) ) {
+			return $value;
+		}
+
+		$new_data = array(
+			'id'           => 'sitekit.withgoogle.com/' . self::SLUG,
+			'slug'         => self::SLUG,
+			'plugin'       => self::SLUG,
+			'new_version'  => $data['version'],
+			'url'          => $data['url'],
+			'package'      => $data['download_url'],
+			'tested'       => $data['tested'],
+			'requires'     => $data['requires'],
+			'requires_php' => $data['requires_php'],
+		);
+		if ( ! empty( $data['icons'] ) ) {
+			$new_data['icons'] = $data['icons'];
+		}
+		if ( ! empty( $data['banners'] ) ) {
+			$new_data['banners'] = $data['banners'];
+		}
+		if ( ! empty( $data['banners_rtl'] ) ) {
+			$new_data['banners_rtl'] = $data['banners_rtl'];
+		}
+
+		$value->response[ self::SLUG ] = (object) $new_data;
+
+		return $value;
+	}
+
+	/**
+	 * Gets plugin data from the API.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array|false Associative array of plugin data, or false on failure.
+	 */
+	private function fetch_plugin_data() {
+		$data = get_site_transient( self::TRANSIENT );
+
+		// Query the Site Kit REST API if the transient is expired.
+		if ( empty( $data ) ) {
+			// phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.wp_remote_get_wp_remote_get
+			$response = wp_remote_get( 'https://sitekit.withgoogle.com/service/dev-plugin-updates/' );
+
+			// Retrieve data from the body and decode json format.
+			$data = json_decode( wp_remote_retrieve_body( $response ), true );
+
+			// Stop here if there is an error, set a temporary transient and bail out.
+			if ( is_wp_error( $response ) || isset( $data['error'] ) || ! isset( $data['version'] ) ) {
+				set_site_transient( self::TRANSIENT, array( 'version' => '0.3.0' ), 30 * MINUTE_IN_SECONDS );
+				return false;
+			}
+
+			set_site_transient( self::TRANSIENT, $data, DAY_IN_SECONDS );
```

Comment:
If this is only called by the `plugin_info` method (see other comments), I think we can get rid of the transient. This request would only be made if someone specifically requested information on the developer plugin, so we can always make it and ensure fresh data - plus one less transient.

In this case we could add a comment on why there is no caching the response here.